package/MDAnalysis/topology/PDBParser.py

```diff
@@ -335,6 +341,30 @@ def _parseatoms(self):
                     validated_elements.append('')
             attrs.append(Elements(np.array(validated_elements, dtype=object)))
 
+        if any(formalcharges):
+            for i, entry in enumerate(formalcharges):
+                if not entry == '':
+                    if entry == '0':
+                        # Technically a lack of charge shouldn't be in the PDB
+                        # but MDA has a few files that specifically have 0
+                        # entries, indicating that some folks interpret 0 as
+                        # an allowed entry
+                        formalcharges[i] = 0
+                    elif ('+' in entry) or ('-' in entry):
+                        try:
+                            formalcharges[i] = int(entry[::-1])
+                        except ValueError:
+                            errmsg = (f"Unknown formal charge {entry} "
+                                      "encountered")
+                            raise ValueError(errmsg)
+                    else:
+                        errmsg = (f"Formal charge {entry} is unrecognized")
+                        raise ValueError(errmsg)
+                else:
+                    formalcharges[i] = 0
+            attrs.append(
+                    FormalCharges(np.array(formalcharges, dtype=np.int32)))
```

Comment:
See above question about `int` vs `int32`

Reply:
Switching to `int` here since the topologyattr code will just re-type to this anyways unless dtype is set to None.

That being said, charges really shouldn't exceed the bounds of int8, so one wonders how wasteful this is... (can't remember if int32/int64 still has the weird behaviour of ending up faster than int8/int16 in some weird cases)